### PR TITLE
<amp-consent> add machtedGeoGroups to request

### PR DESCRIPTION
**Pull request by micajuine-ho**
Task 5 of #25623.
 Add `matchedGeoGroups` to the the checkConsentHref request request body.

Previous task(s):
1: #25676 (merged)
2: #25689 (merged)
3 & 4: #25696 (merged)

### CODE REVIEW

File: extensions/amp-consent/0.1/amp-consent.js [hunk 1]
```diff
@@ -132,6 +135,8 @@ export class AmpConsent extends AMP.BaseElement {
     // ConsentConfig has verified that there's one and only one consent instance
     this.consentId_ = this.consentConfig_['consentInstanceId'];
 
+    this.matchedGeoGroups_ = this.consentConfig_['matchedGeoGroups'];
```
Comment (lannka) [inline extensions/amp-consent/0.1/amp-consent.js]: it's a bit weird to get that from the merged `config` JSON.
`ConsentConfig` should have a public method to return that.

Comment (micajuine-ho) [reply extensions/amp-consent/0.1/amp-consent.js]: Should we also do the same for `consentInstanceId `? Or does it make sense to pull that directly from the config? I figure the latter.

File: extensions/amp-consent/0.1/amp-consent.js [hunk 2]
```diff
@@ -100,6 +100,9 @@ export class AmpConsent extends AMP.BaseElement {
 
     /** @private {?string} */
     this.consentId_ = null;
+
+    /** @private {?string} */
+    this.matchedGeoGroups_ = null;
```
Comment (lannka) [inline extensions/amp-consent/0.1/amp-consent.js]: should only match one group

this.matchedGeoGroup_

File: extensions/amp-consent/0.1/amp-consent.js [hunk 3]
```diff
@@ -115,13 +118,26 @@ export class AmpConsent extends AMP.BaseElement {
       'amp-consent should have an id'
     );
 
-    const config = new ConsentConfig(this.element);
+    const configManager = new ConsentConfig(this.element);
+
+    configManager.getConsentConfigPromise().then(validatedConfig => {
+      this.initialize_(validatedConfig, configManager);
+    });
```
Comment (lannka) [inline extensions/amp-consent/0.1/amp-consent.js]: you can get matchedGeoGroup_ here

File: extensions/amp-consent/0.1/consent-config.js [hunk 4]
```diff
@@ -42,6 +42,9 @@ export class ConsentConfig {
     /** @private {!Window} */
     this.win_ = toWin(element.ownerDocument.defaultView);
 
+    /** @private {?string} */
+    this.matchedGeoGroups_ = null;
```
Comment (lannka) [inline extensions/amp-consent/0.1/consent-config.js]: singular

File: extensions/amp-consent/0.1/consent-config.js [hunk 5]
```diff
@@ -51,12 +54,21 @@ export class ConsentConfig {
    * @return {!Promise<!JsonObject>}
    */
   getConsentConfigPromise() {
-    if (!this.configPromise_) {
+    if (!this.config_) {
```
Comment (lannka) [inline extensions/amp-consent/0.1/consent-config.js]: bad merge?

File: extensions/amp-consent/0.1/amp-consent.js [hunk 6]
```diff
@@ -532,6 +536,7 @@ export class AmpConsent extends AMP.BaseElement {
           'consentStateValue': getConsentStateValue(storedInfo['consentState']),
           'consentString': storedInfo['consentString'],
           'isDirty': !!storedInfo['isDirty'],
+          'matchedGeoGroups': this.matchedGeoGroup_,
```
Comment (lannka) [inline extensions/amp-consent/0.1/amp-consent.js]: singular

File: extensions/amp-consent/0.1/consent-config.js (hunk 5 above)
Comment (micajuine-ho) [reply extensions/amp-consent/0.1/consent-config.js]: yeah, seems like it.

File: extensions/amp-consent/0.1/amp-consent.js [hunk 7]
```diff
@@ -441,15 +445,13 @@ export class AmpConsent extends AMP.BaseElement {
   getConsentRequiredPromise_() {
     if (!isExperimentOn(this.win, 'amp-consent-geo-override')) {
       return this.getConsentRequiredPromiseLegacy_();
-    }
-    if (typeof this.consentConfig_['consentRequired'] === 'boolean') {
+    } else if (typeof this.consentConfig_['consentRequired'] === 'boolean') {
```
Comment (lannka) [inline extensions/amp-consent/0.1/amp-consent.js]: `else` not needed

File: extensions/amp-consent/0.1/consent-config.js [hunk 8]
```diff
@@ -57,6 +60,15 @@ export class ConsentConfig {
     return this.configPromise_;
   }
 
+  /**
+   * Returns the matched geoGroup. Call after getConsentConfigPromise
+   * has resolved.
+   * @return {?string}
+   */
+  getMatchedGeoGroups() {
```
Comment (lannka) [inline extensions/amp-consent/0.1/consent-config.js]: singular

File: extensions/amp-consent/0.1/test/test-amp-consent.js [hunk 9]
```diff
@@ -230,6 +242,29 @@ describes.realWin(
         expect(await ampConsent.getConsentRequiredPromise_()).to.be.true;
       });
 
+      it('send post request to server with matched group', async () => {
```
Comment (lannka) [inline extensions/amp-consent/0.1/test/test-amp-consent.js]: add a test case that when multiple groups are matched, we send the first one.

**Pull request merged**